compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeNarrower.java

```diff
@@ -264,8 +266,13 @@ public void visit(BLangTypeTestExpr typeTestExpr) {
             // Terminate for undefined symbols
             return;
         }
+
         final TypeChecker.AnalyzerData data = new TypeChecker.AnalyzerData();
         data.env = env;
+        // If typeNode has finite types with unary expressions in the value space, we need to
+        // convert them into numeric literals.
+        semanticAnalyzer.analyzeNode(typeTestExpr.typeNode, data.env);
+
```

Comment:
Not the correct place IMO. The semantic analyzer/type checker should ideally handle this in their visits.